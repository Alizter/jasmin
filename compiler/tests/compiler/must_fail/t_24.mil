// fcond must be bool (FIXME: would be better to point also to decl)

extern fn test(x : reg u64) -> stack u64 {
  y : stack u64;
  y = 42;
  do {
    y += y;
  } while x;
  return x;
}

/*
START:CMD
ARG="renumber_fun_unique,typecheck"
END:CMD
*/
