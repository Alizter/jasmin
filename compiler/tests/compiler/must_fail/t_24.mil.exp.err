Parsing tests/compiler/must_fail/t_24.mil as MIL

tests/compiler/must_fail/t_24.mil:8:11: 8:12 error: type mismatch (got u64, expected bool)
  } while x;
          ^^
