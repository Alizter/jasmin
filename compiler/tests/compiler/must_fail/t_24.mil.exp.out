renumbering identifiers in all function(s)
   REMOVED_TIME
type checking module in all function(s)
